web/client/components/geostory/contents/Text.jsx

```diff
@@ -29,7 +48,24 @@ export default compose(
     branch(
         ({ mode }) => mode === Modes.EDIT,
         editableText
-    )
-
-
+    ),
+    /*
+     * pass the placeholder localized to the text component
+     * when contentEditing is false (the only case the placeholder is visible)
+    */
+    branch(
```

Comment:
I suggest to : 
 - Avoid branch (it's already checked inside withProps) instead use withProps(onChange if you want to optimize) and do not pass placeholder otherwise. (hoping localizedProps skips if didn't find the string to localize).

Reply:
ok for removing branch, not about the rest.
localizedProps does not skip it returns the original text, but if no value is passed there is the msgId.split problem

Reply:
Looks strange. 
```
 [propName]: props[propName] && getMessage(messages, props[propName])
```
This should prevent to execute getMessage if props["placeholder"] doesn't exist
Can you debug it?

Reply:
apparently text is never rendered if content Editing is true, so non need to perform checks with it, becsuse of the renderComponent inside the enhancer, anyway you are right about localized props skipping undefined values